docs/dev-guide/golang/versioning.md

```diff
@@ -0,0 +1,168 @@
+
+You only need to preserve the first call to `GetVersion()` for each `changeID`. All subsequent calls to
+`GetVersion()` with the same change Id are safe to remove. If necessary, you can remove the first
+`GetVersion()` call, but you need to ensure the following:
+
+- All executions with an older version are completed.
+- You can no longer use `Step1` for the changeId. If you need to make changes to that same part in
+  the future, such as change from ActivityD to ActivityE, you would need to use a different changeId
+  like `Step1-fix2`, and start minVersion from DefaultVersion again. The code would look like the
+  following:
+
+```go
+v := workflow.GetVersion(ctx, "Step1-fix2", workflow.DefaultVersion, 1)
+if v == workflow.DefaultVersion {
+        err = workflow.ExecuteActivity(ctx, ActivityD, data).Get(ctx, &result1)
+} else {
+        err = workflow.ExecuteActivity(ctx, ActivityE, data).Get(ctx, &result1)
+}
+```
+
+Upgrading a Workflow is straightforward if you don't need to preserve your currently running Workflow Executions.
+You can simply terminate all of the currently running Workflow Executions and suspend new ones from being created while you deploy the new version of your Workflow code, which does not use `GetVersion()`, and then resume Workflow creation.
+However, that is often not the case, and you need to take care of the currently running Workflow Executions, so using `GetVersion()` to update your code is the method to use.
+
+However, if you want your currently running Workflows to proceed based on the current Workflow logic,
+but you want to ensure new Workflows are running on new logic, you can define your Workflow as a
+new `WorkflowType`, and change your start path (calls to `StartWorkflow()`) to start the new Workflow
+type.
+
+## Sanity checking
+
+The Temporal client SDK performs a sanity check to help prevent obvious incompatible changes.
+The sanity check verifies whether a Command made in replay matches the event recorded in history,
+in the same order. The Command is generated by calling any of the following methods:
+
+- workflow.ExecuteActivity()
+- workflow.ExecuteChildWorkflow()
+- workflow.NewTimer()
+- workflow.RequestCancelWorkflow()
+- workflow.SideEffect()
+- workflow.SignalExternalWorkflow()
+- workflow.Sleep()
+
+Adding, removing, or reordering any of the above methods triggers the sanity check and results in
+a non-deterministic error.
+
+The sanity check does not perform a thorough check. For example, it does not check on the Activity's
+input arguments or the timer duration. If the check is enforced on every property, then it becomes
+too restricted and harder to maintain the Workflow code. For example, if you move your Activity code
+from one package to another package, that changes the `ActivityType`, which technically becomes a different
+Activity. But, we don't want to fail on that change, so we only check the function name part of the
+`ActivityType`.
```

Comment:
Moving activity code form one package to another doesn't change the activity type as we just use the function name. So the checker passes as the name is the same.